addons/analytic/models/analytic_plan.py

```diff
@@ -147,15 +147,26 @@ def _compute_analytic_account_count(self):
 
     @api.depends('account_ids', 'children_ids')
     def _compute_all_analytic_account_count(self):
-        plans_count = dict(
-            self.env['account.analytic.account']._read_group(
-                domain=[('root_plan_id', 'in', self.ids)],
-                aggregates=['id:count'],
-                groupby=['root_plan_id']
-            )
+        # Get all children_ids from each plan
+        self.env.cr.execute("""
+            SELECT parent.id,
+                   array_agg(child.id) as children_ids
+              FROM account_analytic_plan parent
+              JOIN account_analytic_plan child ON child.parent_path LIKE parent.parent_path || '%%'
+             WHERE parent.id IN %s
+          GROUP BY parent.id
+        """, [tuple(self.ids)])
+        all_children_ids = self.env.cr.dictfetchall()
+
+        plans_count = self.env['account.analytic.account']._read_group(
+            domain=[('plan_id', 'child_of', self.ids)],
+            aggregates=['id:count'],
+            groupby=['plan_id']
         )
-        for plan in self:
-            plan.all_account_count = plans_count.get(plan, 0)
+        plans_count = {p[0].id: p[1] for p in plans_count}
+
+        for plan, all_children_ids in zip(self, all_children_ids):
```

Comment:
This is dangerous, if you don't have children, there will be no value for that plan, leading to `len(self) != len(all_children_ids)`.
Also, you have no order on `all_children_ids`, you would need to sort both the SQL query and `self`.

It would be safer to just
```python
        self.env.cr.execute(...)
        plan2children = dict(self.env.cr.fetchall())
        plan2count = dict(self.env['account.analytic.account']._read_group(...))
        for plan in self:
            plan.all_account_count = sum(plan2count[child_id] for child_id in plan2children.get(plan.id, [])
```

Reply:
> if you don't have children, there will be no value for that plan
`all_children_ids` still contains the plan id itself, no?

Anyway, we still have the order problem. What you're suggest could work but I need either 
1) the keys of `plans_count` to be an `id` not a record by doing `plans_count = {k.id: v for k, v in plans_count.items()}`
2) browse the children plans inside the loop

I prefer the 1st option but I'd would like to have your opinion before pushing?